src/go/rpk/pkg/cli/registry/subject.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+)
+
+func subjectCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "subject",
+		Args:  cobra.ExactArgs(0),
+		Short: "List or delete schema registry subjects",
+	}
+	cmd.AddCommand(
+		subjectListCommand(fs, p),
+		subjectDeleteCommand(fs, p),
+	)
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+func subjectListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var deleted bool
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "Display all subjects",
+		Args:    cobra.ExactArgs(0),
+		Run: func(*cobra.Command, []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]string{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewSRClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			subjects, err := cl.Subjects(context.Background(), sr.HideShowDeleted(deleted))
+			out.MaybeDieErr(err)
+
+			if isText, _, s, err := f.Format(subjects); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+			}
+			for _, s := range subjects {
+				fmt.Println(s)
+			}
+		},
+	}
+	cmd.Flags().BoolVar(&deleted, "deleted", false, "If true, list deleted subjects as well")
+	return cmd
+}
+
+type deleteResponse struct {
```

Comment:
For what it's worth, the types that are only used within a single function can be defined within the function itself.

Reply:
Yes, I just don't want to stuff everything at the beginning of the command `Run` function and I tried to leave "only the logic" we would end up with:

```
		Run: func(cmd *cobra.Command, subjects []string) {
			type schemaResponse struct {
				Subject string `json:"subject" yaml:"subject"`
				Version int    `json:"version,omitempty" yaml:"version,omitempty"`
				ID      int    `json:"id,omitempty" yaml:"ID,omitempty"`
				Type    string `json:"type,omitempty" yaml:"type,omitempty"`
				Err     string `json:"error,omitempty" yaml:"error,omitempty"`
			}
			f := p.Formatter
			if h, ok := f.Help([]schemaResponse{}); ok {
				out.Exit(h)
			}
```

Before even starting with the logic.

I know it can be inside of the `newListCommand` function but if it's ok for you, I'll leave it outside.

Reply:
Putting it inside newDeleteCommand is what I meant, but I'm not too attached.